File: src/main/java/com/dozycoffee/wms/global/error/ErrorResponseDto.java
package com.dozycoffee.wms.global.error;

import java.time.LocalDateTime;

public record ErrorResponseDto(String errorCode, String message, LocalDateTime timestamp) {

    public static ErrorResponseDto of(ErrorCode errorCode) {
        return new ErrorResponseDto(errorCode.getCode(), errorCode.getMessage(), LocalDateTime.now());
    }
}

File: src/main/kotlin/com/dozycoffee/wms/global/error/ErrorResponseDto.kt
package com.dozycoffee.wms.global.error

import java.time.LocalDateTime

data class ErrorResponseDto(
    val errorCode: String,
    val message: String,
    val timestamp: LocalDateTime
) {
    companion object {
        fun of(errorCode: ErrorCode): ErrorResponseDto {
            return ErrorResponseDto(errorCode.code, errorCode.message, LocalDateTime.now())
        }
    }
}
